src/promptflow-evals/promptflow/evals/evaluate/_eval_run.py

```diff
@@ -273,51 +313,83 @@ def log_artifact(self, artifact_folder: str) -> None:
+        account_url = f"{datastore.account_name}.blob.{datastore.endpoint}"
+        svc_client = BlobServiceClient(
+            account_url=account_url, credential=self._get_datastore_credential(datastore))
+        for local, remote in zip(local_paths, remote_paths['paths']):
+            blob_client = svc_client.get_blob_client(
+                container=datastore.container_name,
+                blob=remote['path'])
+            with open(local, 'rb') as fp:
+                blob_client.upload_blob(fp, overwrite=True)
+
+        # To show artifact in UI we will need to register it. If it is a promptflow run,
+        # we are rewriting already registered artifact and need to skip this step.
+        if self._is_promptflow_run:
+            return
+        url = (
+            f"https://{self._url_base}/artifact/v2.0/subscriptions/{self._subscription_id}"
+            f"/resourceGroups/{self._resource_group_name}/providers/"
+            f"Microsoft.MachineLearningServices/workspaces/{self._workspace_name}/artifacts/register"
+        )
+
         response = self.request_with_retry(
-            url=self.get_artifacts_uri(), method="POST", json_dict=remote_paths, headers=headers
+            url=url,
+            method="POST",
+            json_dict={
+                "origin": "ExperimentRun",
+                "container": f"dcid.{self.info.run_id}",
+                "path": EvalRun.EVALUATION_ARTIFACT,
+                "dataPath": {
+                    "dataStoreName": datastore.name,
+                    "relativePath": posixpath.join(root_upload_path, EvalRun.EVALUATION_ARTIFACT),
+                },
+            },
         )
         if response.status_code != 200:
-            self._log_error("allocate Blob for the artifact", response)
-            return
-        empty_artifacts = response.json()["artifactContentInformation"]
-        # The response from Azure contains the URL with SAS, that allows to upload file to the
-        # artifact store.
-        for local, remote in zip(local_paths, remote_paths["paths"]):
-            artifact_loc = empty_artifacts[remote["path"]]
-            blob_client = BlobClient.from_blob_url(artifact_loc["contentUri"], max_single_put_size=32 * 1024 * 1024)
-            with open(local, "rb") as fp:
-                blob_client.upload_blob(fp)
+            self._log_warning('register artifact', response)
+
+    def _get_datastore_credential(self, datastore: Datastore):
+        # Reference the logic in azure.ai.ml._artifact._artifact_utilities
+        # https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/ml/azure-ai-ml/azure/ai/ml/_artifacts/_artifact_utilities.py#L103
+        credential = datastore.credentials
+        if isinstance(credential, AccountKeyConfiguration):
+            return credential.account_key
+        elif hasattr(credential, "sas_token"):
+            return credential.sas_token
+        else:
+            return self._ml_client.datastores._credential

```

Comment:
We should be able to get rid of ml_client since this is being used only for datastore cred looks like

Reply:
We use ML_client in several places
 - To [get auth token](https://github.com/microsoft/promptflow/blob/7b6495800aaf22f9e40537452127a8612c97d383/src/promptflow-evals/promptflow/evals/evaluate/_eval_run.py#L253C47-L253C57)
 - To [get datastore](https://github.com/microsoft/promptflow/blob/7b6495800aaf22f9e40537452127a8612c97d383/src/promptflow-evals/promptflow/evals/evaluate/_eval_run.py#L346)
 - And, finally, to [get the datastore credential](https://github.com/microsoft/promptflow/blob/7b6495800aaf22f9e40537452127a8612c97d383/src/promptflow-evals/promptflow/evals/evaluate/_eval_run.py#L392C20-L392C58).